weather-api-widget/weather.lua

```diff
@@ -364,6 +366,181 @@ local function worker(user_args)
         end
     }
 
+    local hourly_forecast_graph = wibox.widget {
+        step_width = 12,
+        color = '#EBCB8B',
+        background_color = beautiful.bg_normal,
+        forced_height = 100,
+        forced_width = 300,
+        widget = wibox.widget.graph,
+        set_max_value = function(self, new_max_value)
+            self.max_value = new_max_value
+        end,
+        set_min_value = function(self, new_min_value)
+            self.min_value = new_min_value
+        end,
+    }
+
+    local hourly_forecast_negative_graph = wibox.widget {
+        step_width = 12,
+        color = '#5E81AC',
+        background_color = beautiful.bg_normal,
+        forced_height = 100,
+        forced_width = 300,
+        widget = wibox.widget.graph,
+        set_max_value = function(self, new_max_value)
+            self.max_value = new_max_value
+        end,
+        set_min_value = function(self, new_min_value)
+            self.min_value = new_min_value
+        end,
+    }
+
+    local hourly_forecast_widget = {
+        layout = wibox.layout.fixed.vertical,
+        update = function(self, hourly)
+            local hours_below = {
+                id = 'hours',
+                forced_width = 300,
+                layout = wibox.layout.flex.horizontal
+            }
+            local temp_below = {
+                id = 'temp',
+                forced_width = 300,
+                layout = wibox.layout.flex.horizontal
+            }
+
+            local max_temp = -1000
+            local min_temp = 1000
+            local values= {}
+
+            -- Yeah, this looks weird. I would expect to have to use ipairs
+            for i, hour in pairs(hourly) do
+                if i > 25 then
+                    break
+                end
+
+                values[i] = hour.temp_c
+
+                if max_temp < hour.temp_c then
+                    max_temp = hour.temp_c
+                end
+
+                if min_temp > hour.temp_c then
+                    min_temp = hour.temp_c
+                end
+
+                if (i - 1) % 5 == 0 then
+                    table.insert(hours_below, wibox.widget {
+                        text = os.date(time_format_12h and '%I%p' or '%H:00', tonumber(hour.time_epoch)),
+                        align = 'center',
+                        font = font_name .. ' 9',
+                        widget = wibox.widget.textbox
+                    })
+
+                    table.insert(temp_below, wibox.widget {
+                        markup = '<span foreground=""'
```

Comment:
Typo? Extra double quote results syntax error
 ```
 markup = '<span foreground="'
```

This works.

Reply:
Overly clever NeoVim, but yes.